### PR TITLE
[server] replace is-reachable with fetch

### PR DESCRIPTION
**Pull request by AlexTugarev**
Getting rid of unnecessary dependency.

## Related Issue(s)

Fixes #

## How to test
* use preview env
  * try creating a new Git Integration and see if the reachability checks still works

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - at-rm-is-reachable</li>
	<li><b>🔗 URL</b> - <a href="https://at-rm-is-reachable.preview.gitpod-dev.com/workspaces" target="_blank">at-rm-is-reachable.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - at-rm-is-reachable-gha.12380</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

**Comment by easyCZ** [comment]
Should be possible to remove `is-reachable` as well, no?

**Comment by easyCZ** [comment]
Now it is, in the previous version it wasn't updating yarn.lock

**Comment by geropl** [comment]
Build-golfing again :golfing_man: : https://github.com/gitpod-io/gitpod/actions/runs/5420453262/jobs/9855039476

File: components/server/src/auth/auth-provider-service.ts [hunk 1]
```diff
@@ -244,7 +244,11 @@ export class AuthProviderService {
         return this.config.hostUrl.with({ pathname }).toString();
     };
 
-    async isHostReachable(host: string) {
-        return await isReachable(host, { timeout: 2000 });
+    async isHostReachable(host: string): Promise<boolean> {
+        try {
+            const resp = await fetch(`https://${host}`, { timeout: 2000, method: "HEAD" });
+            return resp.ok;
+        } catch (error) {}
```
Comment (easyCZ) [inline components/server/src/auth/auth-provider-service.ts:251]: Let's log this out. I think it can help us in debugging some aspects of dedciated

Comment (geropl) [reply components/server/src/auth/auth-provider-service.ts:251]: Separate PR please. I need the yarn.lock changes to continue :pray:

**Comment by geropl** [comment]
@AlexTugarev There are compatibility issues: https://github.com/gitpod-io/gitpod/actions/runs/5420453262/jobs/9855249827#step:14:3092

Do you want to check those?

**Comment by geropl** [comment]
image
@AlexTugarev

**Comment by AlexTugarev** [comment]
Fix is building. 
TIL; _HEAD_ is not always supported
```
> (await fetch("https://gitlab.gitpod-dev.com/", { method: "HEAD"})).status
404
```

**Comment by geropl** [comment]
Host checked passed now, seeing another error (with my config I guess): 
image

**Review by geropl — APPROVED** [review]
:heavy_check_mark:

**Comment by geropl** [comment]
@AlexTugarev un-hold?

**Comment by AlexTugarev** [comment]
Thanks for the review, @easyCZ and @geropl!

File: components/server/src/auth/auth-provider-service.ts [hunk 2]
```diff
@@ -244,7 +244,13 @@ export class AuthProviderService {
...
     };
 
-    async isHostReachable(host: string) {
-        return await isReachable(host, { timeout: 2000 });
+    async isHostReachable(host: string): Promise<boolean> {
+        try {
+            const resp = await fetch(`https://${host}`, { timeout: 2000 });
+            return resp.ok;
+        } catch (error) {
+            console.log(`Host is not reachable: ${host}`);
```
Comment (easyCZ) [inline components/server/src/auth/auth-provider-service.ts:252]: FYI, should use `log.warn` or similar

**Pull request merged**